### PR TITLE
feat: Add import reader for binlog

### PR DESCRIPTION
**Pull request by bigsheeper**
This PR defines the new import reader interfaces and implement a binlog reader for import.

issue: https://github.com/milvus-io/milvus/issues/28521

### CODE REVIEW

File: internal/util/importutilv2/binlog/util.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	}
+	var reader *storage.BinlogReader
+	reader, err = storage.NewBinlogReader(bytes)
+	if err != nil {
+		return nil, merr.WrapErrImportFailed(fmt.Sprintf("failed to create reader, binlog:%s, error:%v", path, err))
+	}
+	return reader, nil
+}
+
+func listBinlogs(cm storage.ChunkManager, paths []string) (map[int64][]string, []string, error) {
```
Comment (czs007) [inline internal/util/importutilv2/binlog/util.go:68]: break into two functions?

File: internal/util/importutilv2/reader.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	"github.com/milvus-io/milvus/internal/storage"
+	"github.com/milvus-io/milvus/internal/util/importutilv2/binlog"
+)
+
+type Reader interface {
+	Next(count int64) (*storage.InsertData, error)
+	Close()
+}
+
+type ColumnReader interface {
```
Comment (czs007) [inline internal/util/importutilv2/reader.go:30]: FieldReader?

File: internal/util/importutilv2/reader.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	Next(count int64) (storage.FieldData, error)
```
Comment (czs007) [inline internal/util/importutilv2/reader.go:31]: The term "Next(count)" has inconsistent semantics, especially when it comes to JSON and binlog readers, as it carries different meanings. I suggest renaming it and removing the "count" parameter. Instead, consider using a parameter that reflects the specific configuration of the reader object during its construction.

File: internal/util/importutilv2/reader.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	Close()
+}
+
+func NewReader(cm storage.ChunkManager,
+	schema *schemapb.CollectionSchema,
+	paths []string,
+	options Options,
```
Comment (czs007) [inline internal/util/importutilv2/reader.go:38]: Introduce an additional parameter to indicate the amount of data to be read in each iteration. When this parameter is not provided, it signifies that the entire file will be read in one go.

File: internal/util/importutilv2/binlog/reader.go [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+		deleteFilter, err := FilterWithDelete(r)
+		if err != nil {
+			return err
+		}
+		r.filters = append(r.filters, deleteFilter)
+	}
+	return nil
+}
+
+func (r *reader) ReadDelete(deltaLogs []string, tsStart, tsEnd uint64) (*storage.DeleteData, error) {
```
Comment (czs007) [inline internal/util/importutilv2/binlog/reader.go:84]: renmae to readDelete

File: internal/util/importutilv2/binlog/reader.go [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+		}
+		fieldData, err := cr.Next(count)
+		if err != nil {
+			cr.Close()
+			return nil, err
+		}
+		cr.Close()
+		insertData.Data[field.GetFieldID()] = fieldData
+	}
+	insertData, err = r.Filter(insertData)
```
Comment (czs007) [inline internal/util/importutilv2/binlog/reader.go:137]: Add a function that first attempts to determine if filtering will occur. If it is determined that filtering will not take place, the function can directly return the previously obtained `insertData`, thus avoiding unnecessary copying.

File: internal/util/importutilv2/binlog/reader.go [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+	if err != nil {
+		return nil, err
+	}
+	r.readIdx++
+	return insertData, nil
+}
+
+func (r *reader) Filter(insertData *storage.InsertData) (*storage.InsertData, error) {
```
Comment (czs007) [inline internal/util/importutilv2/binlog/reader.go:145]: Add a step first attempts to determine if filtering will occur. If it is determined that filtering will not take place, the function can directly return the previously obtained insertData, thus avoiding unnecessary copying.

File: internal/util/importutilv2/binlog/column_reader.go [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+	if err != nil {
+		return nil, err
+	}
+	return &columnReader{
+		reader:      reader,
+		fieldSchema: fieldSchema,
+	}, nil
+}
+
+func (r *columnReader) Next(_ int64) (storage.FieldData, error) {
```
Comment (czs007) [inline internal/util/importutilv2/binlog/column_reader.go:40]: It is important to note that if there is no data in "insertData" after filtering, returning an "insertData" with 0 rows may be mistakenly interpreted by the upper layer as all data being read.

**Comment by bigsheeper** [comment]
rerun ut

File: internal/util/importutilv2/binlog/reader.go [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+		field := typeutil.GetField(r.schema, fieldID)
+		if field == nil {
+			return nil, merr.WrapErrFieldNotFound(fieldID)
+		}
+		path := binlogs[r.readIdx]
+		fr, err := newFieldReader(r.cm, field, path)
+		if err != nil {
+			return nil, err
+		}
+		fieldData, err := fr.Next(-1)
```
Comment (czs007) [inline internal/util/importutilv2/binlog/reader.go:143]: fr.Next() ?
you can modify this in next pr?

**Pull request merged**